### PR TITLE
Feature/delete index

### PR DESCRIPTION
**Pull request by polishq**
This PR solves several issues related to the "custom route" functionality, mainly aimed at preventing stores from accidentally adding duplicate entries to their sitemap:

- It is now impossible to use the `saveIndex` mutation to add an index that already exists
- A new `deleteIndex` mutation is provided to allow custom routes to be removed
- When building the sitemap, any duplicate extended indexes that exist from prior usage will be filtered out

New version is linked here: https://arthursitemap--sandboxusdev.myvtex.com/admin/graphql-ide
And sitemap can be checked here: https://arthursitemap--sandboxusdev.myvtex.com/sitemap.xml

### CODE REVIEW

**Review by ArthurTriis1 — APPROVED** [review]
LGTM! Thanks for contribute!

File: node/resolvers/saveIndex.ts [hunk 1]
```diff
@@ -1,18 +1,41 @@
...
+  const { index: extendedIndexes } = (await vbase.getJSON<SitemapIndex>(
+    bucket,
+    EXTENDED_INDEX_FILE,
+    true
+  )) || { index: [] }
+  // check if index is already in the extendedIndexes
+  if (!extendedIndexes.includes(index)) {
+    extendedIndexes.push(index)
+  }
+  await vbase
```
Comment (lariciamota) [inline node/resolvers/saveIndex.ts:30]: If it is already on the extendedIndexes, should we be running this `saveJSON` anyway?

File: node/resolvers/deleteIndex.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { EXTENDED_INDEX_FILE, getBucket, hashString } from '../utils'
```
Comment (lariciamota) [inline node/resolvers/deleteIndex.ts:1]: I've noticed that this file is almost the same as the `saveIndex` one, do you think we could refactor it to reuse code? Like have a function to get `extendedIndexes` and another to save, so in each resolver, we would only call them and do what is different - for example, filter in the delete case.

**Review by icazevedo — APPROVED** [review]
LGTM

File: node/resolvers/saveIndex.ts (hunk 1 above)
Comment (polishq) [reply node/resolvers/saveIndex.ts:30]: I think it's still worth running the `saveJSON` in order to eliminate any duplicate indexes that may have been added previously. Does that make sense?

Comment (lariciamota) [reply node/resolvers/saveIndex.ts:30]: Oh, now I understand what you did there. 
Wouldn't the duplicates be eliminated in the next sitemap generation (the changes you made on `sitemap.ts`)?

File: node/resolvers/deleteIndex.ts (hunk 2 above)
Comment (polishq) [reply node/resolvers/deleteIndex.ts]: @lariciamota Done! I also updated my workspace link with these changes.

File: node/resolvers/saveIndex.ts (hunk 1 above)
Comment (polishq) [reply node/resolvers/saveIndex.ts]: Well, in `sitemap.ts` the indexes are loaded from vbase, then any duplicates are removed, then the sitemap is generated. But the code doesn't save the de-duped indexes back to vbase, so they still exist.

**Pull request merged**